From 8b7283c836cd357439501789bbbfb9fdd324c137 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:08:52 -0400
Subject: [powerpc] irq: Use generic_handle_irq

Message-id: <1410545655-205645-304-git-send-email-dzickus@redhat.com>
Patchwork-id: 94334
O-Subject: [RHEL7 PATCH 303/626] powerpc: Irq: Use generic_handle_irq
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit a4e04c9f219d2c00764ffa7ba45500411815879d
Author: Thomas Gleixner <tglx@linutronix.de>
Date:   Sun Feb 23 21:40:08 2014 +0000

    powerpc: Irq: Use generic_handle_irq

    No functional change

    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Benjamin Herrenschmidt <benh@kernel.crashing.org>
    Cc: ppc <linuxppc-dev@lists.ozlabs.org>
    Link: http://lkml.kernel.org/r/20140223212736.333718121@linutronix.de
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/kernel/irq.c b/arch/powerpc/kernel/irq.c
index 6296ff5..464a4a3 100644
--- a/arch/powerpc/kernel/irq.c
+++ b/arch/powerpc/kernel/irq.c
@@ -465,7 +465,6 @@ static inline void check_stack_overflow(void)
 
 void __do_irq(struct pt_regs *regs)
 {
- struct irq_desc *desc;
  unsigned int irq;
 
  irq_enter();
@@ -487,11 +486,8 @@ void __do_irq(struct pt_regs *regs)
  /* And finally process it */
  if (unlikely(irq == NO_IRQ))
   __get_cpu_var(irq_stat).spurious_irqs++;
- else {
-  desc = irq_to_desc(irq);
-  if (likely(desc))
-   desc->handle_irq(irq, desc);
- }
+ else
+  generic_handle_irq(irq);
 
  trace_irq_exit(regs);
 
-- 
1.7.1